scripts/deployCollector.ts

```diff
@@ -0,0 +1,159 @@
+export const DEFAULT_OUTPUT_FILE_NAME = 'revenue-sharing-addresses.json';
+
+export interface Partner {
+  beneficiary: string;
+  share: number;
+}
+
+interface CollectorConfig {
+  collectorOwner: string;
+  partners: Partner[];
+  tokenAddress: string;
+  remainderAddress: string;
+}
+
+export type DeployCollectorArg = {
+  collectorConfigFileName?: string;
+  outputFileName?: string;
+};
+
+type ChainId = string;
+type OutputCollectorConfig = CollectorConfig | { collectorContract: string };
+export type OutputConfig = Record<ChainId, OutputCollectorConfig>;
+
+export const deployCollector = async (
+  taskArgs: DeployCollectorArg,
+  hre: HardhatRuntimeEnvironment
+) => {
+  const { ethers, network } = hre;
+
+  if (!network) {
+    throw new Error('Unknown Network');
+  }
+
+  let collectorConfigFileName = taskArgs.collectorConfigFileName;
+
+  if (!collectorConfigFileName) {
+    console.warn(
+      "Missing '--collectorConfig' parameter, 'deploy-collector.input.json' will be used"
+    );
+    collectorConfigFileName = DEFAULT_CONFIG_FILE_NAME;
+  }
+
+  if (!fs.existsSync(collectorConfigFileName)) {
+    throw new Error('Could not find Collector configuration file');
+  }
+
+  const inputConfig = JSON.parse(
+    fs.readFileSync(collectorConfigFileName, { encoding: 'utf-8' })
+  ) as CollectorConfig;
+
+  const { collectorOwner, partners, tokenAddress, remainderAddress } =
+    inputConfig;
+
+  const collectorFactory = await ethers.getContractFactory('Collector');
+  const collector = await collectorFactory.deploy(
+    collectorOwner,
+    tokenAddress,
+    partners,
+    remainderAddress
+  );
+
+  await printReceipt(collector);
+
+  console.log();
+
+  console.log(
+    '|=============================================|==================================================|'
+  );
+  console.log(
+    '| Entity                                      | Address                                          |'
+  );
+  console.log(
+    '|=============================================|==================================================|'
+  );
+  partners.forEach(function (partner, i) {
+    console.log(
+      `| Revenue Partner #${i + 1}, share` +
+        ' '.repeat(20 - (i + 1).toString().length) +
+        `| ${partner.beneficiary}, ${partner.share}%` +
+        ' '.repeat(4 - partner.share.toString().length) +
+        `|`
+    );
+  });
+  console.log(
+    `| Collector Contract                          | ${collector.address}       |`
+  );
+  console.log(
+    `| Collector Owner                             | ${await collector.owner()}       |`
+  );
+  console.log(
+    `| Collector Token                             | ${await collector.token()}       |`
+  );
+  console.log(
+    `| Collector Remainder                         | ${remainderAddress}       |`
+  );
+  console.log(
+    '|=============================================|==================================================|\n'
+  );
```

Comment:
why not use `console.table`?

Reply:
Because of the `partners` field. Since its value is an array, the result is very ugly, because it considers each value of the array as a column, so we end up having a table with 4 very large columns, not so readable IMHO.

Reply:
Agreed on the usage of console.table(), please have a look at this: https://github.com/rsksmart/rif-relay-contracts/blob/feature/PP-278/PP-329/upgrade_contracts/scripts/deploy.ts#L141

Reply:
Yes, I managed to solve the issue I had with the partners by creating an object specifically to be printed.